### PR TITLE
FizzBuzzShashwat.java

### PR DESCRIPTION
**Pull request by shashwatxdevelop**
Please make sure you have done the following:

- [x] Created a folder with the name of language you code inside `/Day<n>/` directory (Where n is for nth day)
- [x] Added your Codes
- [x] Tested your codes
- [x] Added credits (author and date) at the top of the code
- [x] Updated README.md
- [ ] Added yourself as a contributor using `all-contributors-cli`

### CODE REVIEW

**Comment by shashwatxdevelop** [comment]
This is my Day 2 of java code.

**Review by aaditkamat — CHANGES_REQUESTED** [review]
Can you split your solutions into seperate PRs based on the day? Also you have added a duplicate Java file in the root of the repository for your Day 1 solution.

**Comment by shashwatxdevelop** [comment]
> Can you split your solutions into seperate PRs based on the day? Also you have added a duplicate Java file in the root of the repository for your Day 1 solution.

Hey! I am really sorry but I am a beginner in github. I don't know how to separate the PRs. Can you guide me through it please?

**Comment by aaditkamat** [comment]
> Hey! I am really sorry but I am a beginner in github. I don't know how to separate the PRs. Can you guide me through it please?

No problem Shashwat. If you want to do this programmatically, you can try exploring the GitHub project [ghstack](https://github.com/ezyang/ghstack). However, given your level of experience, I would recommend against doing that. Away to do this manually is to checkout different branches from your forked repo's master branch and make changes specific to a particular day in the branch labelled for the day. For example, you could have 3 seperate branches: `day-1`, `day-2` and `day-3` with the solution for the respective day and then send 3 PRs. 

Anyways, from what I see, most people have combined solutions from different days in a single PR, so I'll approve this to save you the trouble. Just make sure to remove the duplicate Java file at the root of the repository.

**Comment by shashwatxdevelop** [comment]
> > Hey! I am really sorry but I am a beginner in github. I don't know how to separate the PRs. Can you guide me through it please?
> 
> No problem Shashwat. If you want to do this programmatically, you can try exploring the GitHub project [ghstack](https://github.com/ezyang/ghstack). However, given your level of experience, I would recommend against doing that. Away to do this manually is to checkout different branches from your forked repo's master branch and make changes specific to a particular day in the branch labelled for the day. For example, you could have 3 seperate branches: `day-1`, `day-2` and `day-3` with the solution for the respective day and then send 3 PRs.
> 
> Anyways, from what I see, most people have combined solutions from different days in a single PR, so I'll approve this to save you the trouble. Just make sure to remove the duplicate Java file at the root of the repository.

Thank you so much Aadit. I have deleted the file you told DayOneShashwat.java. Please see if it is deleted or not. When I click on the file it shows that this file is deleted. Please check and tell me if its done. If not, I will try it again. Thank you once again!

File: Day3/Java/HammingDistance1.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		s1 = sc.nextLine();  //inputing string 1
+		s2 = sc.nextLine();  // imputing string 2
+		l1 = s1.length();    // finding length of string 1
+		l2 = s2.length();    // finding length of string 2
+		
+		if(l1==l2)        // comparing both lengths of strings because hamming distance works only on same length of string
+		{
+			for (i=0 ; i<l1; i++)   // for loop for extraction of characters from both of the strings
+			{ 
+				if (s1.charAt(i)== s2.charAt(i))  // checking same character of both strings
```
Comment (aaditkamat) [inline Day3/Java/HammingDistance1.java:19]: This if condition is redundant.
```suggestion
				if (s1.charAt(i)!= s2.charAt(i))  // checking same character of both strings
				   k += 1;
```

File: Day3/Java/HammingDistance1.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+					k = k;                       // if same character is found no change in counter variable k 
+				else
```
Comment (aaditkamat) [inline Day3/Java/HammingDistance1.java:21]: Remove this else condition

File: Day2/README.md [hunk 3]
```diff
@@ -876,6 +876,34 @@ require './reverse'
...
+		char ch = 0;
+         Scanner sc= new Scanner (System.in);
+         s = sc.nextLine();   // Input the string
+         l = s.length();     // Taking the length of string
+         for(i=l-1; i>=0; i--)       // For loop for the extraction of character from the string s
+         {
+        	 ch = s.charAt(i);
+        	 s1 = s1+ ch;               // Storing the extracted character in new string s1 
+         }
+          System.out.println(s1);             // Printing the new reversed string s1
```
Comment (aaditkamat) [inline Day2/README.md:899]: You can remove the print statement for the new reversed string

File: Day3/README.md [hunk 4]
```diff
@@ -354,7 +354,38 @@ k=[i for i,j in zip(c,d) if i!=j]
...
+		s1 = sc.nextLine();  //inputing string 1
+		s2 = sc.nextLine();  // imputing string 2
+		l1 = s1.length();    // finding length of string 1
+		l2 = s2.length();    // finding length of string 2
+		
+		if(l1==l2)        // comparing both lengths of strings because hamming distance works only on same length of string
+		{
+			for (i=0 ; i<l1; i++)   // for loop for extraction of characters from both of the strings
+			{ 
+				if (s1.charAt(i)== s2.charAt(i))  // checking same character of both strings
```
Comment (aaditkamat) [inline Day3/README.md:376]: You can update the README as per the suggestions for the Java source code file.

**Review by aaditkamat — CHANGES_REQUESTED** [review]
Thanks for deleting DayOneShashwat.java from the repository root. I've suggested some small changes for your Day2 and Day3 solutions. Other than that, your PR looks good to go!

**Review by shashwatxdevelop — COMMENTED** [review]
Hello Aadit! I have made the changes told by you. Please check it once and tell me if I have done it right and everything is okay.

File: Day3/README.md [hunk 5]
```diff
@@ -354,7 +354,37 @@ k=[i for i,j in zip(c,d) if i!=j]
...
+		if(l1==l2)        // comparing both lengths of strings because hamming distance works only on same length of string
+		{
+			for (i=0 ; i<l1; i++)   // for loop for extraction of characters from both of the strings
+			{ 
+				if (s1.charAt(i)!= s2.charAt(i))  // checking same character of both strings
+					k+ = 1;                       // if same character is not found change in variable k by 1
+				
+			}
+		}
+		System.out.println("Hamming Distance is " + k);   // printing the hamming distance
```
Comment (aaditkamat) [inline Day3/README.md:381]: Prints out that Hamming Distance is 1 when the lengths of the strings are unequal in place of displaying an error message. There should be an else clause to handle that case. Sorry I missed that out in my previous review. Do test your code with sample inputs (and perhaps your own test cases) to make sure it matches with the accepted solution.

**Review by aaditkamat — CHANGES_REQUESTED** [review]
I have suggested one more change that I missed out in my last review.

**Review by shashwatxdevelop — COMMENTED** [review]
Hello Aadit! I have made the required changes. Please review them. Also I want to ask one thing that from Day 4 should I start making branches or just continue to make a single PR?

File: Day3/Java/HammingDistance1.java [hunk 6]
```diff
@@ -20,8 +20,13 @@ public static void main(String[] args) {
 					k+ = 1;                       // if same character is not found change in counter variable k by 1
 				
 			}
-		}
+		
 		System.out.println("Hamming Distance is " + k);   // printing the hamming distance
+		}
+		else
+		{
+		System.out.println("Hamming Distance is 1"); // if both strings are unequal then hamming distance is eaqual to 1
```
Comment (aaditkamat) [inline Day3/Java/HammingDistance1.java:28]: Hamming Distance is undefined for strings of unequal length. You can check the official solution [here](https://github.com/CodeToExpress/dailycodebase/blob/master/Day3/JavaScript/hamming.js).
```suggestion
		System.err.println("Hamming Distance is undefined"); // if both strings are unequal then hamming distance is undefined
```

**Review by aaditkamat — CHANGES_REQUESTED** [review]

**Comment by aaditkamat** [comment]
Hi @shashwatxdevelop, I would recommend making seperate branches from your fork's master for each day's solution and send the PR from each respective branch.

**Review by shashwatxdevelop — COMMENTED** [review]
I have made the changes in Day 3 code. Please check it.

**Comment by aaditkamat** [comment]
> I have made the changes in Day 3 code. Please check it.

The comment on the line still says that the Hamming Distance for two strings of unequal length is 1. Please update that.

**Review by shashwatxdevelop — COMMENTED** [review]
I made the changes in the comment and well as in the day 3 readme file. Please check that.

**Review by aaditkamat — APPROVED** [review]
Excellent! Thank you for contributing.

**Pull request merged**